pkg/config/sysprep.go

```diff
@@ -66,31 +66,40 @@ func validateAutounattendPresence(dirPath string) error {
 // CreateSysprepDisks creates Sysprep iso disks which are attached to vmis from either ConfigMap or Secret as a source
 func CreateSysprepDisks(vmi *v1.VirtualMachineInstance) error {
 	for _, volume := range vmi.Spec.Volumes {
-		if volume.Sysprep != nil {
-			if sysprepVolumeHasContents(*volume.Sysprep) {
-				var filesPath []string
-
-				sysprepSourcePath := GetSysprepSourcePath(volume.Name)
-
-				if err := validateAutounattendPresence(sysprepSourcePath); err != nil {
-					return err
-				}
-
-				filesPath, err := getFilesLayout(sysprepSourcePath)
-				if err != nil {
-					return err
-				}
-
-				disk := GetSysprepDiskPath(volume.Name)
-				if err := createIsoConfigImage(disk, sysprepVolumeLabel, filesPath); err != nil {
-					return err
-				}
-
-				if err := ephemeraldiskutils.DefaultOwnershipManager.SetFileOwnership(disk); err != nil {
-					return err
-				}
-			}
+		if !shouldCreateSysprepDisk(volume) {
+			continue
 		}
+		if err := createSysprepDisk(volume); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func shouldCreateSysprepDisk(volume v1.Volume) bool {
```

Comment:
To avoid unnecessary copy of the whole volume, I would pass to this method just `volume.Sysprep` and not the whole volume.